shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1848,6 +1874,14 @@ - (BOOL)hasText {
 }
 
 - (void)insertText:(NSString*)text {
+  NSString* emoji = GetEmoji(text);
+  if (emoji) {
+    self.lastEmoji = emoji;
+  }
+  if (text.length == 1 && !text.UTF8String &&
```

Comment:
We only do this workaround when the text length is 1 which is how the bug is happening in my test cases.